evdns.c

```diff
@@ -4891,7 +4891,7 @@ evdns_base_new(struct event_base *event_base, int flags)
 #else
 		r = evdns_base_resolv_conf_parse(base, opts, "/etc/resolv.conf");
 #endif
-		if (r) {
+		if (r && (6 != r)) {
```

Comment:
LGTM, but let's replace this magic number `6` with some const.

Reply:
And also it seems that it is pretty easy to add a unit test (`test/regress_dns.c`) for this case?

Reply:
Do you have a good location for the const?  I'm assuming in include/event/dns.h where the return values are called out for the function?

```
  If this function encounters an error, the possible return values are: 1 =
  failed to open file, 2 = failed to stat file, 3 = file too large, 4 = out of
  memory, 5 = short read from file, 6 = no nameservers listed in the file
```

Reply:
As for the unit test, we would need to mock out the `evutil_read_file` of "/etc/resolv.conf" since the path is hard coded into evdns_base_new with no option to provide an alternate file.  There is a test `dns_nameservers_no_default_test` that I could add a check for evdns_base_resolv_conf_parse of "empty-resolv.conf" to make sure it returns 6 (or the new const value)

Reply:
>Do you have a good location for the const? I'm assuming in include/event/dns.h where the return values are called out for the function?

Yep, it should be putted into the `dns.h`.

Reply:
>As for the unit test, we would need to mock out the evutil_read_file of "/etc/resolv.conf" since the path is hard coded into evdns_base_new with no option to provide an alternate file.

Ok, let's not bother with the mocking.

> There is a test dns_nameservers_no_default_test that I could add a check for evdns_base_resolv_conf_parse of "empty-resolv.conf" to make sure it returns 6 (or the new const value)

This will be enough.

Reply:
My mistake, it is only called 'empty-resolv.conf' instead it is a missing file so evdns_base_resolv_conf_parse would return 1 (file not found).  I will plan create the file in the test `dns_nameservers_no_default_test` and delete it in the same test.